src/pages/VenuePage/TemplateWrapper.tsx

```diff
@@ -152,7 +153,13 @@ const TemplateWrapper: React.FC<TemplateWrapperProps> = ({ venue }) => {
           hasBackButton={hasBackButton}
         >
           <AnnouncementMessage message={venue.bannerMessage} />
-          {template}
+          {[VenueTemplate.posterhall, VenueTemplate.posterpage].includes(
```

Comment:
It might be worth extracting `[VenueTemplate.posterhall, VenueTemplate.posterpage]` into a `const` outside of the JSX, potentially in `src/settings` where we have similar things defined that are also used in this `.includes()` kind of way.

Reply:
Had the same thought, opted to keep it hardcoded inside that file  since I saw it more connected to the large hardcoded constants `switch`board above.
I could move it or leave it in place, looks the same to me at this point, but should we refactor it into smaller chunks, `if`s, factory functions etc., the settings array would be reusable in at least 2 places, surely it can be moved elsewhere.

Reply:
will leave it as is for now

Reply:
FWIW: test driven the f04735d193f4fcc125643d3277c3f6322378c4c7 and it is nearly perfect for my UX.  Even if scrolling position is not to be cached as well -- it is just a minor aspect. Thank you @goran-peoski-work for all this work!

Reply:
@goran-peoski-work not sure if/what you changed here as you didn’t leave a commit hash, but our existing standard is to extract these arrays to a const in settings as described above, so would make sense to me to follow suit. It’s a different usecase to the ‘switchboard’ defined in this file

Reply:
Also I’m not sure what you mean by ‘refactor into smaller chunks’/ifs/etc. What are you referring to here, and for what purpose?

Reply:
> It’s a different usecase to the ‘switchboard’ defined in this file

Didn't clam it's the same use case, just the same concern: _which templates have randomized posters_, that currently is contained within that one file.

 The original code I wrote didn't change the JSX at the `return` point but baked the Provider within the template. Ultimately that harder to visually track (with fall through cases and if check in), even if it provided for simpler return block.

> What are you referring to here, and for what purpose?

`switch` statements aren't just glorified `goto`s, but can also be considered code smell that something simple like few `if-else`s has been overengineered, or something too complicated hasn't been engineered enough with more robust code using factory functions, single responsibility etc.

@0xdevalias  Changes were made at d9761b2c, please do provide clear yes/no if you still need the array moved to `src/settings`

Reply:
Do you feel like this switch is ‘overengineered’? Doesn’t seem that way to me. And so I’m confused why you bring it up/make those comments here in general?

Reply:
I don’t see how the commit hash your posted above relates to this at all?

Reply:
> Do you feel like this switch is ‘overengineered’?

I don't feel like it. Example of "overengineered" would be a simple `if-else` turned into `switch`, not the case here.

>  And so I’m confused why you bring it up/make those comments here in general?

I can't answer if you're confused, can only why I brought it up here - as an attempted answer to

> What are you referring to here, and for what purpose?

I guess I didn't property address the "purpose", so to be short: **in case it got refactored some day**.

Reply:
Added the constant in d404b6c3244edd64977f6e9cec73f64506bb5fca